api/v1alpha1/types.go

```diff
@@ -0,0 +1,118 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+type ProvisionerID string
+
+// BundleClassSpec defines the desired state of BundleClass
+type BundleClassSpec struct {
+	Provisioner ProvisionerID `json:"provisioner"`
+
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
```

Comment:
Minor detail, and something I'm not sure we really care about given we're very early in the dev stages of this project, but if we're interested in mirroring the StorageClass API, we may want to move these fields outside of the Spec and have everything live in the top-level BundleClass structure.

Reply:
I feel you. Unfortunately, the conventions for CRD-defined types _always_ include a `spec` field. I'm not sure if doing so would break other tooling (e.g. codegen, on-cluster validations, etc.).

Reply:
That's a good point and something that I hadn't considered. Let's hold off for now then and follow-up on this later if we feel the need to revisit.